File: ant-sdk/antd-lua/examples/02-data.lua
--- Example 02: Store and retrieve public data, with cost estimation.
--
-- Prerequisite: antd daemon running on local testnet.

local antd = require("antd")

local client = antd.new_client()

local payload = "Hello, Autonomi!"

-- Estimate cost before storing
local est, err1 = client:data_cost(payload)
if err1 then
    print("Cost error: " .. err1.message)
    os.exit(1)
end
print(string.format(
    "Estimate: %d bytes in %d chunks, storage %s atto, gas %s wei, mode %s",
    est.file_size, est.chunk_count, est.cost, est.estimated_gas_cost_wei, est.payment_mode
))

-- Store public data
local result, err2 = client:data_put_public(payload)
if err2 then
    print("Put error: " .. err2.message)
    os.exit(1)
end
print("Stored at address: " .. result.address)
print(string.format("Chunks stored: %d, payment mode: %s", result.chunks_stored, result.payment_mode_used))

-- Retrieve it back
local retrieved, err3 = client:data_get_public(result.address)
if err3 then
    print("Get error: " .. err3.message)
    os.exit(1)
end
print("Retrieved: " .. retrieved)

assert(retrieved == payload, "Round-trip mismatch!")
print("Public data round-trip OK!")

File: ant-sdk/antd-php/examples/02-data.php
<?php

/**
 * Example 02: Store and retrieve public data, with cost estimation.
 *
 * Prerequisite: antd daemon running on local testnet.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Autonomi\Antd\AntdClient;

$client = new AntdClient();

$payload = 'Hello, Autonomi!';

// Estimate cost before storing
$est = $client->dataCost($payload);
echo "Estimate: {$est->fileSize} bytes in {$est->chunkCount} chunks, "
   . "storage {$est->cost} atto, gas {$est->estimatedGasCostWei} wei, "
   . "mode {$est->paymentMode}\n";

// Store public data
$result = $client->dataPutPublic($payload);
echo "Stored at address: {$result->address}\n";
echo "Chunks stored: {$result->chunksStored}, payment mode: {$result->paymentModeUsed}\n";

// Retrieve it back
$data = $client->dataGetPublic($result->address);
echo "Retrieved: {$data}\n";

if ($data !== $payload) {
    throw new RuntimeException('Round-trip mismatch!');
}
echo "Public data round-trip OK!\n";
